enterprise/server/cmd/executor/executor.go

```diff
@@ -332,7 +332,7 @@ func main() {
 	http.Handle("/readyz", env.GetHealthChecker().ReadinessHandler())
 
 	schedulerOpts := &scheduler_client.Options{}
-	reg, err := scheduler_client.NewRegistration(env, taskScheduler, executorID, schedulerOpts)
+	reg, err := scheduler_client.NewRegistration(env, taskScheduler, executorID, executionServer.Name(), schedulerOpts)
```

Comment:
I think the local variable definition in this function (`executionServer, ... := executor.NewExecutor(...)`) is leading to some confusion -- the local variable would probably be better named `executor`, because we don't really think of executors as servers in the current model (executors act as clients of the app, and the app schedules execution tasks across the registered client executors). We used to use a server model, which is probably why this variable name was originally chosen.

The name `ExecutionServer`, at least internally, typically refers to the gRPC `ExecutionService` implementation (`execution_server.go`) that lives on the app server, which accepts `Execute` requests and then schedules them as tasks to be executed on one of the client executors.

Users don't necessarily know/care about this architecture, but so far we have used the term "executor" to refer to the client executor binaries, both in our docs and in the UI (e.g. "use self-hosted executors" in settings). So I think we should replace the term "execution server name" with "executor name" throughout this PR

Reply:
So the reason I used execution server name instead of executor name is that the instances of the executor binaries already have IDs/names associated with them that are not these names; these names are instead the persistent names associated with the machine an executor runs on. I am concerned that if we call both things "executor names", no one looking at the code will have any idea what the difference between the two is supposed to be.

Reply:
+1 for "executor host" or "executor host ID" like Vadim mentioned in another thread

Reply:
sgtm

Reply:
Fixed.